pkg/controller/ingressconfig/ingressconfig_controller.go

```diff
@@ -0,0 +1,132 @@
+package ingressconfig
+
+import (
+	"context"
+	"log"
+	"time"
+
+	operv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/cluster-network-operator/pkg/controller/statusmanager"
+	"github.com/openshift/cluster-network-operator/pkg/names"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// The periodic resync interval.
+// We will re-run the reconciliation logic, even if the network configuration
+// hasn't changed.
+var ResyncPeriod = 3 * time.Minute
+
+// ManifestPaths is the path to the manifest templates
+// bad, but there's no way to pass configuration to the reconciler right now
+var ManifestPath = "./bindata"
+
+// Add creates a new ingressConfig controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager, status *statusmanager.StatusManager) error {
+	return add(mgr, newIngressConfigReconciler(mgr.GetClient()))
+}
+
+// newIngressConfigReconciler returns a new reconcile.Reconciler
+func newIngressConfigReconciler(client client.Client) *ReconcileIngressConfigs {
+	return &ReconcileIngressConfigs{client: client}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r *ReconcileIngressConfigs) error {
+	// create a controller and register watcher for ingresscontroller resource
+	c, err := controller.New("ingress-config-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+	err = c.Watch(&source.Kind{Type: &operv1.IngressController{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileIngressConfigs{}
+
+// ReconcileIngressConfigs watches for updates to ingress controller configuration
+// and sets the network policy related labels on the openshift-host-network namespace
+type ReconcileIngressConfigs struct {
+	client client.Client
+}
+
+// Reconcile sets the openshift-host-network namespaces' labels as per the
+// endpointPublishingStrategy of the `default` ingress controller object.
```

Comment:
Can you add a bit more description here? What labels are added in response to what settings.